packages/react-search-ui-views/src/SearchBox.js

```diff
@@ -53,25 +56,35 @@ function SearchBox(props) {
                 appendClassName("sui-search-box", className) + autocompleteClass
               }
             >
-              <div className="sui-search-box__wrapper">
-                <input
-                  {...getInputProps({
+              <InputView
+                getInputProps={additionalProps =>
```

Comment:
I don't think I'm understanding where `additionalProps` are coming from or how devs should pass this in in a final usage situation. Is there any chance we could write an extra test that shows a custom `inputView` function with additional props for both the search input and the button input?

Reply:
Ah, duh, I see an example in the Storybook file below. I'd still really like to see a test written for the new functionality possible, but totally get if not or if it's better suited for an integration test.

I'm also a little curious about the purpose of the markdown file examples vs the storybook examples - would it make sense to combine them at one point so all examples are in one location? Or are they specifically split up for a reason?

Reply:
The markdown files are the documentation, meant for people using the project. The storybook is more for us, being able to test and styles components in various states in isolation. I would very much like to consolidate this. I think, ideally, that component reference section could just be replaced with storybook.

Reply:
@constancecchen I didn't realize that we don't have a test for this. It was in severe need. I added tests that cover both `getInputProps` and `getButtonProps`. I also did some reworking so they preserve css class names, which would have been a big issue. Thank you for your keen eyes.

I unfortunately could not find a way to test `getAutocomplete`, but I need to move on from this PR. The smoke tests in storybook and the sandbox will have to be enough for now.

I also added notes about this pattern to the README. I'd like to expand more views, in the future, to use this pattern: https://github.com/downshift-js/downshift#prop-getters

Reply:
All good, this is an amazing amount of extra tests/documentation!! Thanks so much for going above and beyond! 😁